temporal/server_option.go

```diff
@@ -190,3 +190,11 @@ func WithChainedFrontendGrpcInterceptors(
 		s.customInterceptors = interceptors
 	})
 }
+
+// WithCustomerMetricsHandler sets a custom implementation of the metrics.MetricHandler interface
+// metrics.MetricHandler is the base interface for publishing metric events
+func WithCustomerMetricsHandler(handler metrics.MetricHandler) ServerOption {
```

Comment:
Please add deprecated to WithCustomMetricsReporter() and point them to this method. We will remove WithCustomMetricsReporter() probably in v1.19.